### PR TITLE
Introduced currency filter on deposits list page

### LINKED ISSUE #1340: Add deposit currency filter to Deposits page
When an account has multiple settlement currencies available, a dropdown filter should be shown above the deposits table. It will allow the table to be filtered by currency, or for all deposits to be shown ( details paJDYF-13A-p2 ).

When there is a mix of currencies displayed, the summary at the bottom of the table should be removed (because we can't sum the values).

### PR DESCRIPTION
**Pull request by kalessil**
Fixes #1340 

#### Changes proposed in this Pull Request

- Introduce currency filter on the deposits list page (shows up when more than one currency is available)

#### Testing instructions

* Connect the client to the server running locally or in the sandbox with server branch 710-gh-Automattic/woocommerce-payments-server
* Ensure the Stripe account has multiple settlement currencies (or change the currency of a deposit in DB to e.g. 'eur')
* Open deposits list page
** Ensure the filter is presented and changing the currency results in the proper deposits rendered
** Ensure sorting/pagination is preserved while changing the currency (for pagination you'll ned 25+ deposits in your account)
** Ensure the estimated deposits are counted properly in totals (make a test transaction for estimated deposit to appear)

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in the description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/deposits/filters/config.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+export const filters = [
+	{
+		label: __( 'Deposit currency', 'woocommerce-payments' ),
+		param: 'store_currency_is',
+		staticParams: [ 'paged', 'per_page', 'orderby', 'order' ],
```
Comment (jrodger) [inline client/deposits/filters/config.js:10]: The designs for this page include advanced filters. Should we create a new issue to track adding those? Just to make it clear we're not planning to do it as part of this multi-deposit currency work.

File: includes/admin/class-wc-rest-payments-charges-controller.php [hunk 2]
```diff
@@ -42,7 +42,7 @@ public function register_routes() {
 	 * @param WP_REST_Request $request Full data about the request.
 	 */
 	public function get_charge( $request ) {
-		$charge_id = $request->get_params()['charge_id'];
+		$charge_id = $request->get_param( 'charge_id' );
```
Comment (jrodger) [inline includes/admin/class-wc-rest-payments-charges-controller.php:45]: Thanks for updating these 👍

**Review by jrodger — COMMENTED** [review]
Looks good so far, I've just left a few comments. I'll give it a final test while reviewing the server endpoint.

File: client/deposits/filters/config.js (hunk 1 above)
Comment (kalessil) [reply client/deposits/filters/config.js:10]: Yes, we definitely need a ticket for the advanced filters. Will create a new one on Monday.

File: readme.txt [hunk 3]
```diff
@@ -106,6 +106,8 @@ Please note that our support for the checkout block is still experimental and th
 * Add - Download transactions report in CSV.
 * Update - Tweak the connection detection logic.
 * Add - Notification badge next to payments menu.
+* Add - Introduced deposit currency filter for deposits overview page.
+
```
[the remark this answers is not in the document]
Comment (kalessil) [reply readme.txt:110]: Done in d1d64261d7cdf691c80ec8a1e7091374eb29fa10

File: includes/admin/class-wc-rest-payments-transactions-controller.php [hunk 4]
```diff
@@ -124,7 +124,7 @@ private function get_transactions_filters( $request ) {
 				'store_currency_is' => $request->get_param( 'store_currency_is' ),
 				'search'            => $request->get_param( 'search' ),
 			],
-			function ( $filter ) {
+			static function ( $filter ) {
```
[the remark this answers is not in the document]
Comment (kalessil) [reply includes/admin/class-wc-rest-payments-transactions-controller.php:127]: Nope. By placing 'static' you say 'don't bind the closure to the object'. Helps PHP to better scope and gabarge collect

File: client/data/deposits/hooks.js [hunk 5]
```diff
@@ -65,5 +67,26 @@ export const useDeposits = ( {
 				isLoading: isResolving( 'getDeposits', [ query ] ),
 			};
 		},
-		[ paged, perPage, orderby, order ]
+		[ paged, perPage, orderby, order, storeCurrencyIs ]
+	);
+
+export const useDepositsSummary = ( {
+	match,
```
[the remark this answers is not in the document]
Comment (kalessil) [reply client/data/deposits/hooks.js:74]: Correct, I placed it as a reminder that we'll add an advance filter later on.

File: client/deposits/filters/config.js (hunk 1 above)
Comment (kalessil) [reply client/deposits/filters/config.js:10]: https://github.com/Automattic/woocommerce-payments/issues/1545

File: client/deposits/index.js [hunk 6]
```diff
@@ -3,19 +3,18 @@
...
 import Page from 'components/page';
 import { TestModeNotice, topics } from 'components/test-mode-notice';
-import DepositsOverview from './overview';
 import DepositsList from './list';
 
 const DepositsPage = () => {
 	return (
 		<Page>
 			<TestModeNotice topic={ topics.deposits } />
-			<DepositsOverview />
```
[the remark this answers is not in the document]
Comment (kalessil) [reply client/deposits/index.js]: Good point.

I reverted the changes, so we keep the overview for now in be32c11de643368c27efbc54ffc409a3d25d150d
Added a new comment into https://github.com/Automattic/woocommerce-payments/issues/1410 regarding the feature-flag switch.

File: includes/admin/class-wc-rest-payments-transactions-controller.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (jrodger) [reply includes/admin/class-wc-rest-payments-transactions-controller.php:127]: TIL, thanks

**Comment by jrodger** [comment]
Thanks for addressing those points. This looks ready to go once the server changes are deployed!

**Pull request merged**